src/geocat/f2py/dpres_plevel_wrapper.py

```diff
@@ -0,0 +1,193 @@
+                be set to the missing value of `pressure_surface`. If there is no missing value
+                for `pressure_surface` then the missing value will be set to the default for
+                float or double appropriately. If `pressure_top` or `pressure_surface` is between
+                pressure_levels levels then the layer thickness is modifed accordingly.
+                If `pressure_surface` is set to a missing value, all layer thicknesses
+                are set to the appropriate missing value.
+
+                The primary purpose of this function is to return layer thicknesses to be used to
+                weight observations for integrations.
+
+            Examples:
+
+                Example 1: Using dpres_plevel with :class:`xarray.DataArray` input
+
+                .. code-block:: python
+
+                    import numpy as np
+                    import xarray as xr
+                    import geocat.comp
+
+                    # Open a netCDF data file using xarray default engine and load the data stream
+                    ds = xr.open_dataset("./SOME_NETCDF_FILE.nc")
+
+                    # [INPUT] Grid & data info on the source
+                    pressure_surface = ds.PS
+                    pressure_levels = ds.LEV
+                    pressure_top = 0.0
+
+                    # Call the function
+                    result_dp = geocat.comp.dpres_plevel(pressure_levels, pressure_surface, pressure_top)
+            """
+
+    # Apply basic sanity checks on the input data
+    pressure_levels, pressure_surface, pressure_top = sanity_check(pressure_levels,
+                                                                   pressure_surface,
+                                                                   pressure_top
+                                                                   )
+
+    # Inner wrapper call
+    dp = _dpres_plevel(pressure_levels.values,
+                       pressure_surface.values,
+                       pressure_top,
+                       msg_py
+                       )
+
+    # Reshape output based on the dimensionality of pressure_surface
+    if pressure_surface.ndim == 1:
+        dp = dp.reshape(dp.shape[1])
+    elif pressure_surface.ndim == 2:
+        dp = dp.reshape(dp.shape[0] * dp.shape[1], dp.shape[2], dp.shape[3])
+
+    if meta:
+        raise MetaError(
+            "ERROR dpres_plevel: Retention of metadata (other than Xarray.Dataarray.attrs) is not yet supported !")
+
+        # TODO: Retaining possible metadata might be revised in the future
+    else:
+        dp = xr.DataArray(dp, attrs=pressure_surface.attrs)
+
+    return dp
+
+
+# Inner Wrapper _<funcname>()
+# This wrapper basically calls the Fortran function. It also handles
+# transpose of the input and output data as well as missing value
+# representations before and after the Fortran function call.
+
+def _dpres_plevel(plev, psfc, ptop, msg_py):
```

Comment:
This runs with the inner wrapper defined after the outer wrapper? If it does, then this is fine.

Reply:
Yes, it works because this code is part of a package that is being installed under a namespace.

Reply:
Thanks, marking as resolved.